### PR TITLE
Finding faithful friends from facebook

### PR DESCRIPTION
**Pull request by jrundquist**
- Added facebook friend-finding. 
- Added merge oauth accounts if found. 
- Added client-side post-hoc authentication.
- Main menu now a view. 
- User profile pictures now pulled from Facebook / Twitter / LinkedIn / Gravatar with default Gravatar if all other fail
- Shark and Router objects now Model and Router instances respectively.

### CODE REVIEW

File: models/user.js [hunk 1]
```diff
@@ -100,4 +121,48 @@ UserSchema.method('getInvited', function(callback){
   User.find({_id: {$nin: this.friends}, school: this.school, friends: this._id}, callback)
 })
 
+UserSchema.method('isDuplicate', function(userObj){
+  // if (this.school != userObj.school){
+  //   console.log (this.school, userOb)
+  //   return false;
+  // }
```
Comment (Gerg) [inline models/user.js]: rm?

File: public/scripts/backbone-schedule/views/friends-list.coffee [hunk 2]
```diff
@@ -1,35 +1,73 @@
...
-    add_friend: ->
-      Shark.friendsList.add new Friend {firstName: "Bob", lastName: "Smith"}
+    events:
+      'click #friend-list-add-from-facebook': 'addFirendFromFacebook'
+      'click #find-and-add-friends' : 'findAndAddFriends' # Should actuall be a call to a friend finding dialog
 
     render: ->
-      @$el.html $ @friendsListTemplate()
-      @$list = @$el.find('#friends-list-content') if not @$list
+      @$el.html @template(list: Shark.friendsList)
```
Comment (Gerg) [inline public/scripts/backbone-schedule/views/friends-list.coffee]: I don't understand why you would do this.

**Comment by Gerg** [comment]
Other than comments, good

File: public/scripts/backbone-schedule/views/friends-list.coffee (hunk 2 above)
Comment (jrundquist) [inline public/scripts/backbone-schedule/views/friends-list.coffee]: ♫  He's a NUT! He's crazy in the coconut  ♫

**Pull request merged**